### ISSUE TITLE
drop host from a zone

### ISSUE THREAD

**Issue by HarrisChu** [vesoft-inc/nebula#2846]
version: nightly

1. add zone with host
2. create a space
3. drop hosts. `DROP HOST "fpplsn-storaged-2.fpplsn-storaged-headless.default.svc.cluster.local":9779 FROM ZONE z1`

expect result:
1. as the host has part, could not drop host from zone.

actual result:
1. drop successfully
2. there's no more zone infomation.

```
(root@nebula) [(none)]> show create space s1
+-------+----------------------------------------------------------------------------------------------------------------------------------------------+
| Space | Create Space                                                                                                                                 |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------+
| "s1"  | "CREATE SPACE `s1` (partition_num = 4, replica_factor = 1, charset = utf8, collate = utf8_bin, vid_type = INT64, atomic_edge = false) ON g1" |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------+
Got 1 rows (time spent 865/1397 us)

Mon, 13 Sep 2021 05:47:43 UTC

(root@nebula) [(none)]>
(root@nebula) [(none)]>
(root@nebula) [(none)]>
(root@nebula) [(none)]>
(root@nebula) [(none)]>
(root@nebula) [(none)]> show groups
+------+------+
| Name | Zone |
+------+------+
| "g1" | "z1" |
+------+------+
Got 1 rows (time spent 678/1163 us)

Mon, 13 Sep 2021 05:47:45 UTC

(root@nebula) [(none)]> show zones
+------+------+------+
| Name | Host | Port |
+------+------+------+
+------+------+------+
Empty set (time spent 794/1225 us)

Mon, 13 Sep 2021 05:47:47 UTC
```

```
(root@nebula) [s1]> show hosts
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| Host                                                                   | Port | Status   | Leader count | Leader distribution  | Partition distribution |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "fpplsn-storaged-0.fpplsn-storaged-headless.default.svc.cluster.local" | 9779 | "ONLINE" | 0            | "No valid partition" | "No valid partition"   |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "fpplsn-storaged-1.fpplsn-storaged-headless.default.svc.cluster.local" | 9779 | "ONLINE" | 0            | "No valid partition" | "No valid partition"   |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "fpplsn-storaged-2.fpplsn-storaged-headless.default.svc.cluster.local" | 9779 | "ONLINE" | 4            | "s1:4"               | "s1:4"                 |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "fpplsn-storaged-3.fpplsn-storaged-headless.default.svc.cluster.local" | 9779 | "ONLINE" | 0            | "No valid partition" | "No valid partition"   |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "fpplsn-storaged-4.fpplsn-storaged-headless.default.svc.cluster.local" | 9779 | "ONLINE" | 0            | "No valid partition" | "No valid partition"   |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "fpplsn-storaged-5.fpplsn-storaged-headless.default.svc.cluster.local" | 9779 | "ONLINE" | 0            | "No valid partition" | "No valid partition"   |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "fpplsn-storaged-6.fpplsn-storaged-headless.default.svc.cluster.local" | 9779 | "ONLINE" | 0            | "No valid partition" | "No valid partition"   |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
| "Total"                                                                |      |          | 4            | "s1:4"               | "s1:4"                 |
+------------------------------------------------------------------------+------+----------+--------------+----------------------+------------------------+
Got 8 rows (time spent 885/1604 us)
```

**Comment by darionyaphet** [comment]
```
(root@nebula) [(none)]> ADD ZONE z_1 192.168.8.215:55501
Execution succeeded (time spent 3400/4110 us)

Thu, 16 Sep 2021 20:02:18 CST

(root@nebula) [(none)]> ADD GROUP g_1 z_1
Execution succeeded (time spent 2942/3553 us)

Thu, 16 Sep 2021 20:02:36 CST

(root@nebula) [(none)]> CREATE SPACE test_1(partition_num=1, replica_factor=1, vid_type=int64) ON g_1
Execution succeeded (time spent 3190/3833 us)

Thu, 16 Sep 2021 20:02:51 CST

(root@nebula) [(none)]> drop HOST 192.168.8.215:55501 FROM ZONE z_1
[ERROR (-1005)]: Conflict!

Thu, 16 Sep 2021 20:03:32 CST

(root@nebula) [(none)]> show hosts
+-----------------+-------+----------+--------------+---------------------+------------------------+
| Host            | Port  | Status   | Leader count | Leader distribution | Partition distribution |
+-----------------+-------+----------+--------------+---------------------+------------------------+
| "192.168.8.215" | 55501 | "ONLINE" | 1            | "test_1:1"          | "test_1:1"             |
+-----------------+-------+----------+--------------+---------------------+------------------------+
| "Total"         |       |          | 1            | "test_1:1"          | "test_1:1"             |
+-----------------+-------+----------+--------------+---------------------+------------------------+
Got 2 rows (time spent 2239/3130 us)

Thu, 16 Sep 2021 20:03:39 CST
```


fix #2877 https://github.com/vesoft-inc/nebula/pull/2877

**Comment by darionyaphet** [comment]
fixed and close
